iceoryx_utils/test/moduletests/test_cxx_newtype.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/cxx/newtype.hpp"
+#include "test.hpp"
+#include "testutils/compile_test.hpp"
+
+using namespace ::testing;
+using namespace iox;
+using namespace iox::cxx;
+
+template <typename T>
+struct Sut : public T
+{
+    using T::T;
+    using T::operator=; // bring all operator= into scope
+};
+
+static CompileTest compileTest(R"(
+    #include \"iceoryx_utils/cxx/newtype.hpp\"
+
+    template <typename T>
+    struct Sut : public T
+    {
+        using T::T;
+        using T::operator=; // bring all operator= into scope
+    };
+
+    using namespace iox;
+    using namespace iox::cxx;
+)",
+                               {"iceoryx_utils/include"});
+
+
+TEST(NewType, ComparableDoesCompile)
+{
+    Sut<cxx::NewType<int, newtype::ConstructByValueCopy, newtype::Comparable>> a(123), b(456);
+    EXPECT_TRUE(a != b);
+    EXPECT_FALSE(a == b);
+}
+
+TEST(NewType, DISABLED_NoComparableDoesNotCompile)
```

Comment:
Enable?

Reply:
the compile tests needs a little bit more love

Reply:
It is a first possible implementation and usage of compilation template testing. We need to get a feeling for that and therefore I left the test in here to see it in practice but I doubt that this is the final version.